In [1]:
import pandas as pd
from pydub import AudioSegment

/opt/homebrew/Caskroom/miniconda/base/envs/bachelor_thesis/lib/python3.8/site-packages/pydub/utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)


# 1. Re naming of columns in df and cliping of audio files

In [2]:
# Pass time in seconds
def clip_audio_and_save (instance_of_df, extra_name):
    time_start = instance_of_df['File Offset (s)']
    time_duration = instance_of_df['Delta Time (s)']
    file_location = 'data/Sounds/' + instance_of_df['File Name']
    file_name = 'data/Clips/' + extra_name + str(instance_of_df.name) + '.wav'
    second_to_milliseconds = 1000
    audio = AudioSegment.from_wav(file_location)
    clip = audio[time_start*second_to_milliseconds: time_start*second_to_milliseconds + time_duration*second_to_milliseconds]
    clip.export(file_name, format='wav')
    return file_name

def clip_all_df (df, extra_name):
    df['Clip Location'] = df.apply(clip_audio_and_save, axis = 1, extra_name = extra_name)



## Other: non gunshots

In [3]:
other = pd.read_csv('data/other/ecoGuns_SST_Guns_Training-OtherSounds.txt', delimiter="\t")
other = other.drop(
    columns=['View', 'Channel', 'Begin Path', 'site', 'uniqueID', 'Notes', 'Score final', 'ID even', 'type shot',
             'total shots', 'Tags', 'Final', 'Selection'])
other['Delta Time (s)'] = other['End Time (s)'] - other['Begin Time (s)']
other = other.rename(columns= {'Relative Path': 'site', 'Begin File': 'File Name'})
other

Begin Time (s)  End Time (s)  Low Freq (Hz)  High Freq (Hz)  \
0           0.2924       0.81871          3.788         931.818   
1      482989.4590  483000.21920        166.667         965.909   
2      483005.2137  483010.88620        121.212         981.061   
3      483033.2839  483036.61730        117.424         969.697   
4      512169.5652  512171.67050        287.879         973.485   
..             ...           ...            ...             ...   
89     587942.2174  587955.55470        165.316        1249.595   
90     661844.8364  661866.53230        194.489        1021.070   
91     752403.3676  752435.82060        145.867        1103.728   
92     823938.2274  823952.25540        149.289         966.825   
93     431721.8376  431734.73880        191.943        1037.915   

    File Offset (s)                  File Name   site     Notes.1  \
0            0.2924  nn02e_20190130_000000.wav  nn02e  start file   
1        51288.4814  nn05f_20190317_000000.wav  nn05f  biological   
2        51304.2361  nn05f_20190317_000000.wav  nn05f  biological   
3        51332.3063  nn05f_20190317_000000.wav  nn05f  biological   
4        80468.5876  nn05f_20190317_000000.wav  nn05f        tree   
..              ...                        ...    ...         ...   
89       69900.1226  nn07c_20180708_000000.wav  nn07c         NaN   
90       57462.1365  nn02g_20180824_000000.wav  nn02g         NaN   
91       61680.5746  nn06c_20181210_000000.wav  nn06c         NaN   
92       46874.3173  nn05f_20190412_000000.wav  nn05f         NaN   
93          20.8600  nn05f_20190317_000000.wav  nn05f         NaN   

    Delta Time (s)  
0          0.52631  
1         10.76020  
2          5.67250  
3          3.33340  
4          2.10530  
..             ...  
89        13.33730  
90        21.69590  
91        32.45300  
92        14.02800  
93        12.90120  

[94 rows x 9 columns]

In [4]:
clip_all_df(other, 'other')
other.to_csv('data_frames/ecoGuns_SST_Guns_Training-OtherSounds.csv')

From this, I have in total 94  clips,
with a total of 616.36 seconds,
approximately 10 minutes

## Ecoguns: gunshots




In [5]:
ecoguns = pd.read_csv('data/ecoguns/Guns_Training_ecoGuns_SST.txt', delimiter = "\t")
ecoguns = ecoguns.drop(columns= ['view', 'channel', 'begin path', 'duration (s)', 'time', 'distance(m)', 'calibre', 'angle', 'elevation', 'uniqueID', 'date', 'Selection', 'site.1'])
ecoguns['Delta Time (s)'] =  ecoguns['end time'] - ecoguns['begin time']
ecoguns = ecoguns.rename(columns={'begin time': 'Begin Time (s)', 'end time': 'End Time (s)', 'low freq': 'Low Freq (Hz)', 'high freq': 'High Freq (Hz)', 'Begin File': 'File Name'})
ecoguns


Begin Time (s)  End Time (s)  Low Freq (Hz)  High Freq (Hz)  \
0         28839.0174    28839.9729             10             900   
1         28843.1979    28844.2331             10             900   
2         28846.5024    28847.6570             10             900   
3         28852.0366    28853.1514             10             900   
4         28854.6245    28855.6995             10             900   
...              ...           ...            ...             ...   
1090      31262.1188    31263.2336             10             900   
1091      31264.7068    31265.7420             10             900   
1092      31273.5853    31274.7797             10             900   
1093      31280.9908    31282.1454             10             900   
1094      31334.6721    31335.8665             10             900   

                      File Name  File Offset (s)   site  numshots  \
0     mbeli_20180517_000000.wav       28839.0174  mbeli         1   
1     mbeli_20180517_000000.wav       28843.1979  mbeli         1   
2     mbeli_20180517_000000.wav       28846.5024  mbeli         1   
3     mbeli_20180517_000000.wav       28852.0366  mbeli         1   
4     mbeli_20180517_000000.wav       28854.6245  mbeli         1   
...                         ...              ...    ...       ...   
1090  tree1_20180517_000000.wav       31262.1188  tree1         1   
1091  tree1_20180517_000000.wav       31264.7068  tree1         1   
1092  tree1_20180517_000000.wav       31273.5853  tree1         1   
1093  tree1_20180517_000000.wav       31280.9908  tree1         1   
1094  tree1_20180517_000000.wav       31334.6721  tree1         1   

      Delta Time (s)  
0             0.9555  
1             1.0352  
2             1.1546  
3             1.1148  
4             1.0750  
...              ...  
1090          1.1148  
1091          1.0352  
1092          1.1944  
1093          1.1546  
1094          1.1944  

[1095 rows x 9 columns]

In [6]:
clip_all_df(ecoguns, 'ecoguns')
ecoguns.to_csv('data_frames/Guns_Training_ecoGuns_SST.csv')

From this, I have 1095 clips
3308.2 seconds,
approximately 55 minutes

## PNNN: gunshots



In [7]:
pnnn_dep = pd.read_csv('data/pnnn_dep/nn_Grid50_guns_dep1-7_Guns_Training.txt', delimiter = "\t")

pnnn_dep = pnnn_dep.drop(columns= ['View', 'Channel', 'Begin Path', 'uniqueID', 'Notes', 'Score final', 'ID even', 'type shot', 'Selection'])
pnnn_dep['Delta Time (s)'] =  pnnn_dep['End Time (s)'] - pnnn_dep['Begin Time (s)']
pnnn_dep = pnnn_dep.rename(columns={'Begin File': 'File Name'})
pnnn_dep

Begin Time (s)  End Time (s)  Low Freq (Hz)  High Freq (Hz)  \
0     3.802536e+04  3.803098e+04        150.800         623.800   
1     1.126124e+05  1.126163e+05        284.800         854.300   
2     2.179818e+05  2.179943e+05          8.700         908.300   
3     2.179949e+05  2.180001e+05        246.400         739.200   
4     2.180339e+05  2.180391e+05        284.800         854.300   
5     2.615889e+05  2.615927e+05        150.800         623.800   
6     3.908804e+05  3.908832e+05         31.400         836.500   
7     3.909045e+05  3.909082e+05          0.000         634.900   
8     3.909106e+05  3.909126e+05         25.200         786.200   
9     4.827551e+05  4.827595e+05        220.500         670.000   
10    5.288658e+05  5.288706e+05        284.800         854.300   
11    6.352122e+05  6.352180e+05        284.800         854.300   
12    7.590955e+05  7.590989e+05        210.500         648.094   
13    8.404524e+05  8.404559e+05        150.800         623.800   
14    8.404627e+05  8.404665e+05        150.800         623.800   
15    8.404679e+05  8.404721e+05        247.400         484.200   
16    8.405498e+05  8.405545e+05        152.600         651.026   
17    8.405700e+05  8.405749e+05        150.800         623.800   
18    8.911513e+05  8.911550e+05        150.800         623.800   
19    1.019685e+06  1.019689e+06        284.800         854.300   
20    1.107570e+06  1.107575e+06        245.700         634.900   
21    1.152915e+06  1.152921e+06         10.000         500.000   
22    1.249342e+06  1.249347e+06         36.735         934.694   
23    1.249380e+06  1.249385e+06         20.408         571.429   
24    1.249570e+06  1.249575e+06         10.000         500.000   
25    1.249808e+06  1.249812e+06         44.898         620.408   
26    1.324748e+06  1.324753e+06         10.000         788.856   
27    1.409280e+06  1.409286e+06         56.008         621.694   
28    1.528876e+06  1.528881e+06         35.191         947.214   
29    1.633435e+06  1.633441e+06         41.056         623.800   
30    1.641236e+06  1.641239e+06        140.200         730.205   
31    1.641248e+06  1.641253e+06         20.528         727.273   
32    1.646577e+06  1.646581e+06         73.314         920.821   
33    1.798166e+06  1.798170e+06        151.000         624.000   
34    1.869496e+06  1.869499e+06         61.584         788.856   
35    1.869610e+06  1.869614e+06         52.786         624.000   
36    1.870085e+06  1.870088e+06        151.000         624.000   
37    1.951483e+06  1.951488e+06        285.000         854.000   
38    1.952738e+06  1.952741e+06        285.000         854.000   
39    1.998361e+06  1.998365e+06        151.000         624.000   
40    2.074859e+06  2.074863e+06        151.000         624.000   
41    2.194034e+06  2.194039e+06         76.246         854.000   
42    2.260874e+06  2.260877e+06         52.786         624.000   
43    2.343490e+06  2.343494e+06        151.000         624.000   
44    2.502671e+06  2.502675e+06        151.000         624.000   
45    2.564148e+06  2.564152e+06        285.000         854.000   
46    2.621273e+06  2.621279e+06        246.000         635.000   
47    2.743514e+06  2.743519e+06        220.500         670.000   
48    2.837685e+06  2.837688e+06         64.516         624.000   
49    2.899911e+06  2.899916e+06         99.707         797.654   
50    2.978056e+06  2.978062e+06        150.800         623.800   
51    3.079859e+06  3.079862e+06         18.500         458.800   
52    3.178136e+06  3.178141e+06        245.700         634.900   
53    3.252040e+06  3.252045e+06        245.700         634.900   
54    3.342605e+06  3.342613e+06        284.800         854.300   
55    3.424887e+06  3.424895e+06        150.800         623.800   
56    3.532390e+06  3.532395e+06        150.800         623.800   
57    3.555043e+06  3.555047e+06        284.800         854.300   
58    3.675109e+06  3.675115e+06        150.800    

From the Sound_Clips, I have 59 clips
Approximately 5 seconds each, so 295 seconds or 4.9 minutes

From the pnnn_dep I have 59 clips
Approximately 272 seconds, 4 minutes

This gives a total of 118 clips and 8.9 minutes



In [8]:
clip_all_df(pnnn_dep, 'pnnn')
pnnn_dep.to_csv('data_frames/nn_Grid50_guns_dep1-7_Guns_Training.csv')
